### PR TITLE
feat(ue5): add ue5 support

### PR DESCRIPTION
**Pull request by CanisHelix**
Add support for UnrealEditor in place of UE4Editor where applicable and new Libc++ paths.

### CODE REVIEW

File: ue4cli/UE4BuildInterrogator.py [hunk 1]
```diff
@@ -166,7 +166,10 @@ def _getThirdPartyLibs(self, platformIdentifier, configuration):
 		# (Ensure we always perform sentinel file cleanup even when errors occur)
 		try:
 			args = ['-Mode=JsonExport', '-OutputFile=' +jsonFile ] if self.engineVersion['MinorVersion'] >= 22 else ['-gather', '-jsonexport=' + jsonFile, '-SkipBuild']
```
Comment (JJC1138) [inline ue4cli/UE4BuildInterrogator.py:168]: This is needed for `ue4 libs` to work with UE5:

```suggestion
			args = ['-Mode=JsonExport', '-OutputFile=' +jsonFile ] if (self.engineVersion['MajorVersion'] >= 5 or self.engineVersion['MinorVersion'] >= 22) else ['-gather', '-jsonexport=' + jsonFile, '-SkipBuild']
```

**Comment by JJC1138** [comment]
I needed another little patch so that libc++ can be found with the slightly different path it has in UE5:

```diff
diff --git a/ue4cli/UnrealManagerLinux.py b/ue4cli/UnrealManagerLinux.py
index 0d4b728..209f189 100644
--- a/ue4cli/UnrealManagerLinux.py
+++ b/ue4cli/UnrealManagerLinux.py
@@ -4,21 +4,6 @@ from .UnrealManagerUnix import UnrealManagerUnix
 from .Utility import Utility
 import re, os
 
-# Details for libc++
-LIBCXX_LIBDIR = '%UE4_ROOT%/Engine/Source/ThirdParty/Linux/LibCxx/lib/Linux/x86_64-unknown-linux-gnu'
-LIBCXX_DETAILS_OVERRIDE = ThirdPartyLibraryDetails(
-	prefixDirs  = [],
-	includeDirs = [
-		'%UE4_ROOT%/Engine/Source/ThirdParty/Linux/LibCxx/include',
-		'%UE4_ROOT%/Engine/Source/ThirdParty/Linux/LibCxx/include/c++/v1'
-	],
-	linkDirs    = [LIBCXX_LIBDIR],
-	libs        = ['{}/libc++.a'.format(LIBCXX_LIBDIR), '{}/libc++abi.a'.format(LIBCXX_LIBDIR), '-lm', '-lc', '-lgcc_s', '-lgcc'],
-	cxxFlags    = ['-fPIC', '-nostdinc++'],
-	ldFlags     = ['-nodefaultlibs'],
-	cmakeFlags  = []
-)
-
 class UnrealManagerLinux(UnrealManagerUnix):
 	
 	def getPlatformIdentifier(self):
@@ -66,4 +51,21 @@ class UnrealManagerLinux(UnrealManagerUnix):
 		return ['libc++']
 	
 	def _getLibraryOverrides(self):
-		return {'libc++': LIBCXX_DETAILS_OVERRIDE}
+		# Details for libc++
+		osType = 'Unix' if self._getEngineVersionDetails()['MajorVersion'] >= 5 else 'Linux'
+		libCXXRoot = '%UE4_ROOT%/Engine/Source/ThirdParty/{0}/LibCxx'.format(osType)
+		libCXXLibDir = '{}/lib/{}/x86_64-unknown-linux-gnu'.format(libCXXRoot, osType)
+		libCXXDetailsOverride = ThirdPartyLibraryDetails(
+			prefixDirs  = [],
+			includeDirs = [
+				'{}/include'.format(libCXXRoot),
+				'{}/include/c++/v1'.format(libCXXRoot)
+			],
+			linkDirs    = [libCXXLibDir],
+			libs        = ['{}/libc++.a'.format(libCXXLibDir), '{}/libc++abi.a'.format(libCXXLibDir), '-lm', '-lc', '-lgcc_s', '-lgcc'],
+			cxxFlags    = ['-fPIC', '-nostdinc++'],
+			ldFlags     = ['-nodefaultlibs'],
+			cmakeFlags  = []
+		)
+		
+		return {'libc++': libCXXDetailsOverride}
```

**Comment by JJC1138** [comment]
Thanks very much for adding those @CanisHelix!

**Comment by dskvr** [comment]
Team and I have confirmed this branch is functioning as expected. Thanks @CanisHelix!

**Comment by adamrehn** [comment]
@CanisHelix thanks for contributing this! Apologies for taking a while to review it, I've been crazy busy over the past few months and my GitHub backlog has grown far larger than I'd like.

**Pull request merged**